File: src/types/patient.ts
import type { RecommendationItem } from "./icode";

export interface PatientInfo {
  hn: string
  fname: string
  lname: string
  cid: string
  pttype: string
  pttype_name: string
  hipdata_code: string
  dob: string
  sex: 'M' | 'F' | string
  age: number
  recommendations: RecommendationItem[];
}

export interface PatientQuery {
  hn?: string;
  cid?: string;
  fname?: string;
  lname?: string;
}

File: src-tauri/src/models/patient.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct PatientInfo {
    pub hn: String,
    pub fname: String,
    pub lname: String,
    pub cid: String,
    pub pttype: String,
    pub pttype_name: String,
    pub hipdata_code: String,
    pub dob: String, // YYYY-MM-DD
    pub sex: String, // 'M' or 'F' or '1'/'2'
    pub age: i32,
}

#[derive(Debug, Deserialize)]
pub struct PatientQuery {
    pub hn: Option<String>,
    pub cid: Option<String>,
    pub fname: Option<String>,
    pub lname: Option<String>,
}
